mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -415,6 +425,64 @@ class PopulateParamsBase {
     return mlir::success();
   }
 
+  mlir::LogicalResult calculateOutputDerivedParams(InitParams *params,
+                                                   int64_t blockSize,
+                                                   ConvolutionContext &ctx,
+                                                   DerivedOutParams &out) {
+    int64_t cVectorLength = 0;
+    ConvOpType op = ctx.getOpType();
+
+    obtainGemmCVecLen(ctx, cVectorLength);
+    int64_t dataPerThread =
+        (params->gemmMPerBlock * params->gemmNPerBlock) / blockSize;
+    if (!(dataPerThread > 0)) {
+      return failure();
+    }
+
+    // TODO: Allow vectorization group size of 2
+    int64_t vectorizationSize = 4;
+    if (ConvOpType::Conv2DBwdDataOpType == op) {
+      vectorizationSize = 1;
+    }
+
+    if ((cVectorLength > 0) && (dataPerThread % vectorizationSize == 0)) {
+      out.dataPerCopy = gcd(dataPerThread, vectorizationSize);
+    } else {
+      out.dataPerCopy = 1;
+    }
+
+    auto &dimIndexVal = ctx.dimIndexVal;
+    // Find dimensions in which the copy will take place
+    switch (op) {
+    case mlir::miopen::Conv2DOpType:
+      if (dimIndexVal["ko"].first == 4) {
+        out.gemmVectorDim = 1;
```

Comment:
Due to PR #423, gemm dimensions may not be fixed as 3 (GemmK / GemmM / GemmN), but might be 4 (GemmK / GemmM / GemmN / GemmKPack), would it be a better idea to use an enumeration?

Reply:
I don't think that's related - these are GemmG x GemmM x GemmN dimensions in the output, which isn't affected by KPACK. We could introduce an enum for clarity, but then we'd have some inconsistency about when we are/aren't using it.